pr_agent/tools/pr_similar_issue.py

```diff
@@ -148,25 +143,24 @@ async def run(self):
         get_logger().info('Publishing response...')
         similar_issues_str = "### Similar Issues\n___\n\n"
         for i, issue_number_similar in enumerate(relevant_issues_number_list):
-            issue = self.git_provider.repo_obj.get_issue(issue_number_similar)
+            issue = self.git_provider.parse_issue_url_and_get_similar_issues(self.issue_url.split('=')[-1], issue_number_similar)
             title = issue.title
-            url = issue.html_url
-            if relevant_comment_number_list[i] != -1:
-                url = list(issue.get_comments())[relevant_comment_number_list[i]].html_url
+            url = self.git_provider.get_issue_url(issue)
             similar_issues_str += f"{i + 1}. **[{title}]({url})** (score={score_list[i]})\n\n"
         if get_settings().config.publish_output:
-            response = issue_main.create_comment(similar_issues_str)
+            response = self.git_provider.parse_issue_url_and_create_comment(similar_issues_str, self.issue_url.split('=')[-1], original_issue_number)
```

Comment:
**Suggestion:** Consider handling exceptions when creating a comment on an issue. This will make the code more robust and prevent unexpected crashes. [robustness]
```suggestion
            try:
                response = self.git_provider.parse_issue_url_and_create_comment(similar_issues_str, self.issue_url.split('=')[-1], original_issue_number)
            except Exception as e:
                get_logger().exception(f"Failed to create issue comment, error: {e}")
```